frontends/p4/simplifyDefUse.cpp

```diff
@@ -614,6 +1029,18 @@ class FindUninitialized : public Inspector {
                 callee.push_back(table);
             }
         } else if (auto em = mi->to<ExternMethod>()) {
+            auto args = em->expr->arguments;
+            if (args->size() > 0) {
+                auto header = args->at(0);
+                if (em->originalExternType->getName().name == "packet_in" &&
+                    em->method->getName().name == "extract") {
```

Comment:
if you assume that extern methods will potentially set the valid bit of every header you probably don't need this case anymore.